test/refactor_nrepl/ns/suggest_libspecs_test.clj

```diff
@@ -0,0 +1,110 @@
+                                         (vector? x) (mapv (fn [s]
+                                                             (let [files (ns-parser/ns-sym->ns-filenames s)]
+                                                               (cond-> s
+                                                                 files (vary-meta assoc :files files)))))))))
+        c-paths (keys (:clj libspecs))
+        s-paths (keys (:cljs libspecs))
+        libspecs (reduce (fn [libspecs k]
+                           (update-in libspecs [:clj k] (fn [candidates]
+                                                          (mapv (fn [candidate]
+                                                                  (vary-meta candidate assoc :used-from :clj))
+                                                                candidates))))
+                         libspecs
+                         c-paths)]
+    (reduce (fn [libspecs k]
+              (update-in libspecs [:cljs k] (fn [candidates]
+                                              (mapv (fn [candidate]
+                                                      (vary-meta candidate assoc :used-from :cljs))
+                                                    candidates))))
+            libspecs
+            s-paths)))
+
+(deftest suggest-libspecs-response
+  (reset! @#'refactor-nrepl.ns.libspecs/cache {})
+  (are [lib-prefix buffer-lc input-lc preferred-aliases project-libspecs expected]
+       (testing [lib-prefix buffer-lc input-lc preferred-aliases project-libspecs]
+         (is (= expected
+                (sut/suggest-libspecs-response {:lib-prefix lib-prefix
+                                                :buffer-language-context buffer-lc
+                                                :input-language-context input-lc
+                                                :preferred-aliases preferred-aliases
+                                                :namespace-aliases-fn (when (seq project-libspecs)
+                                                                        (constantly (add-file-meta project-libspecs)))})))
+         true)
+    #_lib-prefix #_buffer-lc #_input-lc #_preferred-aliases                  #_project-libspecs              #_expected
+    "test"       "clj"       "clj"      []                                   {}                              ["[clojure.test :as test]"]
+    "test"       "cljs"      "cljs"     []                                   {}                              ["[cljs.test :as test]"]
+    "test"       "cljc"      "clj"      []                                   {}                              ["#?(:clj [clojure.test :as test])"]
+    "test"       "cljc"      "cljs"     []                                   {}                              ["#?(:cljs [cljs.test :as test])"]
+    "test"       "cljc"      "cljc"     []                                   {}                              ["#?(:clj [clojure.test :as test]\n      :cljs [cljs.test :as test])"]
```

Comment:
> At the same time, if an existing file already had an :as test, then test<SLASH> should not trigger cljr-slash?

It should prevent it from triggering if it already exists. That said, I think the client currently doesn't distinguish if the `:as ns-alias` is only for a cljs or clj language context. So it will need to be updated to check if the as alias is active for the language context at the point of call for cljr-slash. 

I think the logic *should be* that if it already exists for the file or language context it should skip cljr-slash, otherwise, it should try cljr-slash on the alias and update the existing reference to either include the missing alias, or amend it to include the language context specific alias.